### PR TITLE
fix: make nextn model inference explicit

### PR DESCRIPTION
**Pull request by jasonqinzhou**
## Summary

- keep the default CLI and Python behavior backward-compatible: omitted `--nextn` / omitted `nextn` disables MTP
- add `--nextn auto` (and Python `nextn=None`) to opt into the model checkpoint/family default
- preserve explicit zero and positive draft depths through task construction and saved experiment configs
- reject negative draft depths and preserve custom acceptance rates during model inference
- rebase onto current `main` after Spica moved out of AIConfigurator in #1389

## Behavior

| Input | Behavior |
|---|---|
| omit `--nextn` | MTP disabled (`nextn: 0`) |
| `--nextn 0` | explicitly disable MTP |
| `--nextn auto` | use `num_nextn_predict_layers`, then the model-family fallback |
| `--nextn N` where `N > 0` | use the explicit draft depth |

The Python API mirrors this contract: `cli_default()` defaults to `nextn=0`; explicitly pass `nextn=None` for model inference.

## Root cause

`TaskV2` uses `None` as its internal model-inference sentinel. The old default CLI path only forwarded truthy positive `nextn` values, so explicit `--nextn 0` was dropped and GLM-5.2 fell back to checkpoint metadata (`num_nextn_predict_layers: 1`). The first PR revision made omission map to that sentinel, but that changed the public default. This revision keeps the public default disabled and makes model inference explicit.

## Validation

- `.venv/bin/python -m pytest -qq -m unit --tb=short --color=no` — 2,220 passed, 39 skipped
- focused CLI/API/serialization tests — 62 passed
- `.venv/bin/ruff check .`
- `.venv/bin/ruff format --check .`
- installed CLI E2E with `nvidia/GLM-5.2-NVFP4`, `gb200`, `vllm 0.11.0`, and `database-mode SOL`:
  - omitted `--nextn`: agg/disagg `exp_config.yaml` contain `nextn: 0`
  - `--nextn 0`: agg/disagg contain `nextn: 0`
  - `--nextn auto`: agg/disagg contain `nextn: 1`

### CODE REVIEW

File: tests/unit/cli/test_nextn_tristate.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            best_configs={},
+            pareto_fronts={"agg": None},
+            tasks=tasks,
+            save_dir=str(tmp_path),
+            backend="vllm",
+        )
+
+    exp_config_path = next(tmp_path.glob("**/agg/exp_config.yaml"))
+    dumped = yaml.safe_load(exp_config_path.read_text(encoding="utf-8"))
+    assert dumped["nextn"] == 0
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply tests/unit/cli/test_nextn_tristate.py:46]: Thanks. This PR does not add or change a generator parameter or backend schema: `nextn` already exists in those interfaces. The change is limited to CLI/API default and sentinel plumbing. Coverage now exercises omitted, explicit zero, and `auto` behavior; public-API forwarding; task construction; and serialized agg/disagg configs. I also ran the installed CLI end to end for all three cases, and the repository e2e suite plus prediction regression gate passed. Since generator-help and schema exposure are unchanged, adding generator-contract tests here would duplicate coverage for an unaffected surface. Closing this thread.

**Comment by jasonqinzhou** [comment]
included in PR 1383.
